src/api/mutator/requester.ts

```diff
@@ -0,0 +1,32 @@
+import { isSSR } from '@dwarvesf/react-utils'
+import Axios, { AxiosRequestConfig } from 'axios'
+import { getToken } from 'context/auth'
+
+// eslint-disable-next-line prefer-destructuring
+const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL
+
+export const AXIOS_INSTANCE = Axios.create({ baseURL: BASE_URL })
+
+export const requester = <T>(config: AxiosRequestConfig): Promise<T> => {
+  const source = Axios.CancelToken.source()
+  // Add token to request header
+  const accessToken = isSSR() ? '' : getToken()
+  if (accessToken) {
+    config.headers = {
+      ...config.headers,
+      Authorization: `Bearer ${accessToken}`,
+    }
+  }
+  const promise = AXIOS_INSTANCE({ ...config, cancelToken: source.token }).then(
+    ({ data }) => data,
+  )
+
+  // @ts-ignore
+  promise.cancel = () => {
+    source.cancel('Query was cancelled by React Query')
```

Comment:
We are using `swr` right?

Reply:
Copied from `swr` example of Orval. Maybe a mistake or a normal term but I was confused too lol. It should be a general word like Axios or requester.